s3stream/src/main/java/com/automq/stream/s3/wal/impl/block/BlockWALService.java

```diff
@@ -273,10 +278,34 @@ private void parseRecordBody(long recoverStartOffset, RecordHeader readRecordHea
 
     @Override
     public WriteAheadLog start() throws IOException {
-        if (started.get()) {
-            LOGGER.warn("block WAL service already started");
-            return this;
+        switch (WAL_STATE.get(this)) {
+            case WAL_STATE_INIT:
+                if (WAL_STATE.compareAndSet(this, WAL_STATE_INIT, WAL_STATE_STARTED)) {
+                    boolean success = false;
+                    try {
+                        doStart();
+                        success = true;
+                    } finally {
+                        if (!success) {
+                            WAL_STATE.compareAndSet(this, WAL_STATE_STARTED, WAL_STATE_INIT);
+                            LOGGER.warn("block WAL service started fail");
+                        }
+                    }
+                }
+                break;
+            case WAL_STATE_STARTED:
+                LOGGER.warn("block WAL service already started");
+                break;
```

Comment:
Currently, the "started" state is primarily used to check whether the metadata required for WAL operation (e.g., WAL Header) is ready (see callers of `checkStarted`), in order to avoid data corruption caused by premature writes. Therefore, the state can only be set to "started" after the `doStart` method executes successfully.
Adding an intermediate "starting" state might resolve the issue.